src/coreclr/src/pal/src/misc/environ.cpp

```diff
@@ -917,44 +917,51 @@ Return Value
     or nullptr otherwise.
 
 --*/
+static char* match_prefix(const char* name, char** list)
+{
+	if(*name == '\0')
+		return nullptr;
+	
+	for (int i = 0; list[i] != nullptr; ++i)
+	{
+		const char* pch = name;
+		char* p = list[i];
+
+		for (;;) {
+			if (*pch == '\0') {
+				if(*p == '=')
+					return p+1;
+					
+				if(*p == '\0') // no = sign -> empty value
+					return p;
+				
+				break;
+			}
+			
+			if (*pch++ != *p++) break;
```

Comment:
A nit - maybe instead of for(;;) and the if with break, we could use do { } while (*pch++ == *p++); here.